### PR TITLE
Bugfix/issue 1377 parcelable

### PR DESCRIPTION
**Pull request by RHenigan**
Fixes #[1377]

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
- [X] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [X] I have run the unit tests with this PR
- [X] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).
- [X] I have tested Android, and Java SE

### Summary
This PR Will:
 - Create SdlPacket and TransportRecord in sdl_android and JavaSE that will extend BaseSdlPacket and BaseTransportRecord respectively
 - Remove Parcelable Methods from BaseSdlPacket and BaseTransportRecord 
 - Deprecate Parcelable Methods in JavaSE SdlPacket and TransportRecord
 - Deprecate base/android/os/Parcel and base/android/os/Parcelable
With these changes we can safely remove base/android/os/Parcel and base/android/os/Parcelable as well as the Parcelable Methods in the JavaSE classes in the next major release

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: javaSE/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+package com.smartdevicelink.transport.utl;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import com.smartdevicelink.transport.enums.TransportType;
+
+public class TransportRecord extends BaseTransportRecord {
+    private TransportType type;
+    private String address;
```
Comment (bilal-alsharifi) [inline javaSE/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java:40]: these vars should be reused from parent class also the constructor should be reused.

File: javaSE/src/main/java/com/smartdevicelink/protocol/SdlPacket.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.smartdevicelink.protocol;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+public class SdlPacket extends BaseSdlPacket {
+    private static final int EXTRA_PARCEL_DATA_LENGTH 			= 24;
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply javaSE/src/main/java/com/smartdevicelink/protocol/SdlPacket.java:38]: removed unused var

File: base/src/main/java/com/smartdevicelink/protocol/BaseSdlPacket.java [hunk 3]
```diff
@@ -47,7 +47,7 @@
  * Any other binder transactions must include an additional int flag into their bundle or the parsing
  * of this object will fail.
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/protocol/BaseSdlPacket.java:48]: removed unused imports

File: base/src/main/java/com/smartdevicelink/transport/utl/BaseTransportRecord.java [hunk 4]
```diff
@@ -32,17 +32,14 @@
...
 
-import android.os.Parcel;
-import android.os.Parcelable;
-
 import com.smartdevicelink.transport.enums.TransportType;
 
-public class TransportRecord implements Parcelable{
+class BaseTransportRecord{
 
     private TransportType type;
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/transport/utl/BaseTransportRecord.java:39]: changed to protected in parent class so child classes have access

File: android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import com.smartdevicelink.transport.enums.TransportType;
+
+public class TransportRecord extends BaseTransportRecord implements Parcelable {
+
+    private TransportType type;
+    private String address;
+
+    public TransportRecord(TransportType transportType, String address) {
+        super(transportType, address);
+        this.type = transportType;
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java:15]: with changing vars to protected child class has access now

File: base/src/main/java/com/smartdevicelink/transport/utl/BaseTransportRecord.java [hunk 6]
```diff
@@ -80,47 +77,4 @@ public String toString(){
         builder.append(address);
         return builder.toString();
     }
-
-    public TransportRecord(Parcel p){
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/transport/utl/BaseTransportRecord.java:84]: reimplemented in base class, added to child classes calling super, Deprecated in javaSE, will need to be moved out of base and into android after cleaning up deprecations

File: android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        dest.writeInt(address !=null? 1 : 0);
+        if(address != null){
+            dest.writeString(address);
+        }
+    }
+
+    public static final Parcelable.Creator<TransportRecord> CREATOR = new Parcelable.Creator<TransportRecord>() {
+        public TransportRecord createFromParcel(Parcel in) {
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java:38]: This was a result of not properly implementing the parcel Constructor, with the other changes I was able to make this match the previous base class implementation

File: javaSE/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java (hunk 1 above)
Comment (RHenigan) [reply javaSE/src/main/java/com/smartdevicelink/transport/utl/TransportRecord.java:40]: resolved after making the vars protected in base class

**Pull request merged**